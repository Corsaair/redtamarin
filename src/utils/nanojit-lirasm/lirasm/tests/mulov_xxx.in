; This Source Code Form is subject to the terms of the Mozilla Public
; License, v. 2.0. If a copy of the MPL was not distributed with this
; file, You can obtain one at http://mozilla.org/MPL/2.0/.

; 46341 * 46341 >= 2^31, and will overflow.
big = immi 46341

; Because 'big' isn't used after mul, it _may_ get allocated to the same
; register as 'res'. This is the case with the ARM back-end, and that is where
; this test is important as rX=rX*rX isn't possible on ARMv5 without some
; trickery.
res = mulxovi big big    ; overflow, so we exit here

; Store 'res' so it isn't dead.
m = allocp 4
sti res m 0
x                       ; we don't exit here

