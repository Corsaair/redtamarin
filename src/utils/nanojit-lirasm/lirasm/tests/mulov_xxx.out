Exited block on line: 12
